[Lateral Movement Detection] Add dashboard and release to production (#4974)
elastic/integrations: packages/lmd/docs/README.md

| Malicious Remote File Creation                | Identifies the file created by a remote host followed by a malware or intrusion detection event triggered by Elastic Endpoint Security.                                                                                            |
| Remote File Creation on a Sensitive Directory | Identifies the file created by a remote host on sensitive directories and folders. Remote file creation in these directories should not be common and could indicate a malicious binary or script trying to compromise the system. |                                                                           |

## Dashboard

The **Lateral Movement Detection Dashboard** is available under **Analytics > Dashboard**. This dashboard gives an overview of anomalies triggered for the lateral movement detection package.

Before going to the dashboard, make sure you have the following settings configured in your Kibana. 
1. You have started the above anomaly detection jobs.
2. After enabling the jobs, go to **Management > Stack Management > Kibana > Data Views**. 

> 3. Click on **Create data view** button and create a data view with an **Index pattern** of **.ml-anomalies***. Make sure you assign the same value to **Name** and **Custom data view ID** under **Show advanced settings**.

Reviewer: As discussed with @sodhikirti07, note that anyone viewing the dashboard will also need read access to the `.ml-anomalies*` index to be able to see the visualizations which query that index directly. To create the data view you also need to enable `Allow hidden and system indices` inside the Advanced settings.

Author: Thank you for catching this. I've made the changes.

| Malicious Remote File Creation                | Identifies the file created by a remote host followed by a malware or intrusion detection event triggered by Elastic Endpoint Security.                                                                                            |
| Remote File Creation on a Sensitive Directory | Identifies the file created by a remote host on sensitive directories and folders. Remote file creation in these directories should not be common and could indicate a malicious binary or script trying to compromise the system. |                                                                           |

## Dashboard

The **Lateral Movement Detection Dashboard** is available under **Analytics > Dashboard**. This dashboard gives an overview of anomalies triggered for the lateral movement detection package.

> Before going to the dashboard, make sure you have the following settings configured in your Kibana. 

Reviewer 2: ```suggestion
For the dashboard to work as expected, the following settings need to be configured in Kibana. 
```

Author: Done

1. You have started the above anomaly detection jobs.
2. You have **read** access to **.ml-anomalies*** index or are assigned **machine_learning_user** role. For more information on roles, please refer to [Built-in roles in Elastic](https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-roles.html).
3. After enabling the jobs, go to **Management > Stack Management > Kibana > Data Views**. 

> 4. Click on **Create data view** button and create a data view with an **Index pattern** of **.ml-anomalies***. Make sure you assign the same value to **Name** and **Custom data view ID** under **Show advanced settings**. You also need to enable **Allow hidden and system indices** inside the **Advanced settings**.

Reviewer 2: Allow hidden and system indices should happen before they can proceed to naming the pattern, so maybe that should be the first sentence in the bullet.

Also maybe explicitly mention what the Name and Data View ID must be, since I'm guessing the dashboard is relying on a specific ID.

Author: Done

| Malicious Remote File Creation                | Identifies the file created by a remote host followed by a malware or intrusion detection event triggered by Elastic Endpoint Security.                                                                                            |
| Remote File Creation on a Sensitive Directory | Identifies the file created by a remote host on sensitive directories and folders. Remote file creation in these directories should not be common and could indicate a malicious binary or script trying to compromise the system. |                                                                           |

## Dashboard

The **Lateral Movement Detection Dashboard** is available under **Analytics > Dashboard**. This dashboard gives an overview of anomalies triggered for the lateral movement detection package.

For the dashboard to work as expected, the following settings need to be configured in Kibana. 
1. You have started the above anomaly detection jobs.

> 2. You have **read** access to **.ml-anomalies*** index or are assigned the **machine_learning_user** role. For more information on roles, please refer to [Built-in roles in Elastic](https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-roles.html).

Reviewer 3: Directly granting access to machine learning results indices has implications for using spaces to control ML job visibility. I think this should be made clear. Something along the lines of this:

```suggestion
2. You have **read** access to **.ml-anomalies*** index or are assigned the **machine_learning_user** role. For more information on roles, please refer to [Built-in roles in Elastic](https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-roles.html). Please be aware that a user who has access to the underlying machine learning results indices can see the results of _all_ jobs in _all_ spaces. Do not grant these permissions if you use Kibana spaces to control which users can see which machine learning results. For more information on machine learning privileges see [here](https://www.elastic.co/guide/en/machine-learning/current/setup.html#setup-privileges).
```

3. After enabling the jobs, go to **Management > Stack Management > Kibana > Data Views**. 
4. Click on **Create data view** button and enable **Allow hidden and system indices** under the **Show Advanced settings**.

> 5. Create a data view with an **Index pattern** of **.ml-anomalies-shared** and assign the same index pattern to **Name** and **Custom data view ID** under **Show advanced settings**.

Reviewer 2: ```suggestion
5. Create a data view with the following settings:
    - Index pattern : `.ml-anomalies-shared`
    - Name: `.ml-anomalies-shared`
    - Custom data view ID: `.ml-anomalies-shared`
```

| Malicious Remote File Creation                | Identifies the file created by a remote host followed by a malware or intrusion detection event triggered by Elastic Endpoint Security.                                                                                            |
| Remote File Creation on a Sensitive Directory | Identifies the file created by a remote host on sensitive directories and folders. Remote file creation in these directories should not be common and could indicate a malicious binary or script trying to compromise the system. |                                                                           |

## Dashboard

The **Lateral Movement Detection Dashboard** is available under **Analytics > Dashboard**. This dashboard gives an overview of anomalies triggered for the lateral movement detection package.

For the dashboard to work as expected, the following settings need to be configured in Kibana. 
1. You have started the above anomaly detection jobs.

> 2. You have **read** access to **.ml-anomalies*** index or are assigned the **machine_learning_user** role. For more information on roles, please refer to [Built-in roles in Elastic](https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-roles.html). Please be aware that a user who has access to the underlying machine learning result's indices can see the results of _all_ jobs in _all_ spaces. Be mindful of granting permissions if you use Kibana spaces to control which users can see which machine learning results. For more information on machine learning privileges, refer to [setup-privileges](https://www.elastic.co/guide/en/machine-learning/current/setup.html#setup-privileges).

Author: @droberts195 Thank for for your suggestion. Please review and lmk if you're okay with the text I added.

The **Lateral Movement Detection Dashboard** is available under **Analytics > Dashboard**. This dashboard gives an overview of anomalies triggered for the lateral movement detection package.

For the dashboard to work as expected, the following settings need to be configured in Kibana. 
1. You have started the above anomaly detection jobs.

> 2. You have **read** access to **.ml-anomalies*** index or are assigned **machine_learning_user** role. For more information on roles, please refer to [Built-in roles in Elastic](https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-roles.html).

Reviewer 4: ```suggestion
2. You have **read** access to **.ml-anomalies*** index or are assigned the **machine_learning_user** role. For more information on roles, please refer to [Built-in roles in Elastic](https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-roles.html).
```
